Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Model/ModelAssetHelpers.h

```diff
@@ -55,5 +104,39 @@ namespace AZ
                 AZStd::span<const float> bitangents,
                 AZStd::span<const float> uvs);
         };
+
+        template<typename T>
+        size_t ModelAssetHelpers::GetAlignedCount(size_t vertexCount, AZ::RHI::Format bufferFormat, uint32_t alignment)
+        {
+            //Calculate vertex data in bytes
+            size_t vertexDataInBytes = vertexCount * RHI::GetFormatSize(bufferFormat);
+
+            //Align up the appropriate alignment. Alignment can be non-power of two.
+            size_t alignedVertexDataInBytes = RHI::AlignUpNPOT(vertexDataInBytes, alignment);
+
+            //Calculate the aligned element count
+            size_t alignedVertexCount = alignedVertexDataInBytes / sizeof(T);
+            return alignedVertexCount;
+        }
+
+        template<typename T>
+        void ModelAssetHelpers::AlignStreamBuffer(
+            AZStd::vector<T>& streamBuffer, size_t vertexCount, AZ::RHI::Format bufferFormat, uint32_t alignment)
+        {
+            size_t alignedCount = GetAlignedCount<T>(vertexCount, bufferFormat, alignment);
+            AZ_Assert(
+                alignedCount >= streamBuffer.size(),
+                "Aligned count should be equal or greater as we are aligning up. Aligned value %i BufferSize %i",
+                alignedCount,
+                streamBuffer.size());
+            size_t alignmentCountDelta = alignedCount - streamBuffer.size();
+
+            // Pad the buffer in order to respect the alignment
+            if (alignmentCountDelta > 0)
+            {
+                AZStd::vector<T> extraIds(alignmentCountDelta, 0);
+                streamBuffer.insert(streamBuffer.end(), extraIds.begin(), extraIds.end());
```

Comment:
On the surface, this looks like a pretty inefficient way to pad the end of the buffer. Isn't this going to cause some unnecessary allocations / deallocations vs something like:
```
for (int i = 0; i < alignmentCountDelta; i++)
{
    streamBuffer.emplace_back(0);
}
```